cache/src/main/java/io/envoyproxy/controlplane/cache/Watch.java

```diff
@@ -60,17 +64,22 @@ public DiscoveryRequest request() {
   }
 
   /**
-   * Sets the callback method to be executed when the watch is cancelled. Even if cancel is executed multiple times, it
-   * ensures that this stop callback is only executed once.
+   * Sends the given response to the watch's response handler.
+   *
+   * @param response the response to be handled
    */
-  public void setStop(Runnable stop) {
-    this.stop = stop;
+  public void respond(Response response) {
+    // TODO: Should trying to respond after cancellation be an error state?
```

Comment:
I'm thinking this should either throw an exception, or log at error level is a response is sent after cancellation. Any preference on which? I'm leaning towards throwing here and handling it higher up since the `DiscoveryServer` has a lot more context that it can include in the log.

Reply:
I'd prefer throwing and handling in the server